CRM/Contribute/BAO/Contribution.php

```diff
@@ -3858,7 +3858,86 @@ public static function recordAdditionalPayment($contributionId, $trxnsData, $pay
     $params['skipLineItem'] = TRUE;
     $trxnsData['trxn_date'] = !empty($trxnsData['trxn_date']) ? $trxnsData['trxn_date'] : date('YmdHis');
     $arAccountId = CRM_Contribute_PseudoConstant::getRelationalFinancialAccount($contributionDAO->financial_type_id, 'Accounts Receivable Account is');
-    if ($paymentType == 'owed') {
+    $fiancialItemPaidStatusID = CRM_Core_PseudoConstant::getKey('CRM_Financial_DAO_FinancialItem', 'status_id', 'Paid');
+    if ($paymentType == 'overpaid') {
+      $params['partial_payment_total'] = $contributionDAO->total_amount + CRM_Utils_Array::value('total_amount', $trxnsData, 0.00);
+      $params['partial_amount_pay'] = CRM_Utils_Array::value('total_amount', $trxnsData, 0.00);
+      $params['pan_truncation'] = CRM_Utils_Array::value('pan_truncation', $trxnsData);
+      $params['card_type_id'] = CRM_Utils_Array::value('card_type_id', $trxnsData);
+      $params['check_number'] = CRM_Utils_Array::value('check_number', $trxnsData);
+      $params['financial_type_id'] = $contributionDAO->financial_type_id;
+      $trxnsData['net_amount'] = !empty($trxnsData['net_amount']) ? $trxnsData['net_amount'] : $trxnsData['total_amount'];
+      $trxnsData['to_financial_account_id'] = CRM_Financial_BAO_FinancialTypeAccount::getInstrumentFinancialAccount($trxnsData['payment_instrument_id']);
+
+      // add financial trxn for the overpaid amount
+      $financialTrxn = CRM_Contribute_BAO_Contribution::recordFinancialAccounts($params, $trxnsData);
+
+      $lineItems = CRM_Price_BAO_LineItem::getLineItemsByContributionID($contributionDAO->id);
+      $lineItemID = NULL;
+      // if line item found then lookout for the one which is configured  with contribution
+      // quick config price field, whose line_total will be updated with overpaid amount.
+      if (!empty($lineItems)) {
+        // find line item params which is used later to store the over paid amount
```

Comment:
NO. The overpaid amount is not to be allocated to line items. Line items can only be paid up to the obligation they create. The overpaid amount is simply not allocated to line items.